six/modules/c++/six.sicd/source/Grid.cpp

```diff
@@ -62,6 +77,381 @@ bool DirectionParameters::operator==(const DirectionParameters& rhs) const
+        {
+            double coef;
+            if (weightType->parameters.empty() || weightType->parameters[0].str().empty())
+            {
+                //A Hamming window is defined in many places as a raised cosine of weight .54,
+                //so this is the default. However, some data use a generalized raised cosine and
+                //call it HAMMING, so we allow for both uses.
+                coef = .54;
+            }
+            else
+            {
+                coef = weightType->parameters[0];
+            }
+
+            weightFunction.reset(new RaisedCos(coef));
+        }
+        else if (windowName == "HANNING")
+        {
+            weightFunction.reset(new RaisedCos(0.50));
+        }
+        else if (windowName == "KAISER")
+        {
+            weightFunction.reset(new Kaiser(weightType->parameters[0]));
+        }
+    }
+
+    return weightFunction;
+}
+
+std::vector<RowColInt>
+DirectionParameters::calculateImageVertices(const ImageData& imageData) const
+{
+    std::vector<RowColInt> vertices;
+
+    if (!imageData.validData.empty())
+    {
+        vertices = imageData.validData;
+    }
+    else
+    {
+        vertices.resize(4);
+        //use edges of full image
+        vertices[0] = RowColInt(0, 0);
+        vertices[1] = RowColInt(imageData.numCols - 1, 0);
+        vertices[2] = RowColInt(imageData.numCols - 1, imageData.numCols - 1);
+        vertices[3] = RowColInt(0, imageData.numCols - 1);
+    }
+    return vertices;
+}
+
+void DirectionParameters::fillDerivedFields(const ImageData& imageData)
+{
+    // Calulating resolution requires fzero and fft functions
+
+    // DeltaK1/2 are approximated from DeltaKCOAPoly
+    if (!Init::isUndefined(deltaKCOAPoly) &&
+        !Init::isUndefined(impulseResponseBandwidth) &&
+        !Init::isUndefined(sampleSpacing) &&
+        Init::isUndefined(deltaK1) &&
+        Init::isUndefined(deltaK2))
+    {
+        // Here, we assume the min and max of DeltaKCOAPoly must be on the vertices
+        // of the image, since it is smooth and monotonic in most cases--although in
+        // actuality this is not always the case. To be totally generic, we would 
+        // have to search for an interior min and max as well.
+        std::pair<double, double> deltas = calculateDeltaKs(imageData);
+        deltaK1 = deltas.first;
+        deltaK2 = deltas.second;
+    }
+
+    if (weightType.get() != NULL &&
+        weights.empty() &&
+        weightType->windowName != "UNKNOWN")
+    {
+        std::auto_ptr<Functor> weightFunction = calculateWeightFunction();
+        if (weightFunction.get())
+        {
+            weights = (*weightFunction)(DEFAULT_WEIGHT_SIZE);
+        }
+    }
+    return;
+}
+
+bool DirectionParameters::validate(const ImageData& imageData,
+    logging::Logger& log) const
+{
+    bool valid = true;
+    std::ostringstream messageBuilder;
+    const double& epsilon = std::numeric_limits<double>::epsilon();

```

Comment:
I don't think it'll matter much performance-wise either way, but in general, making a reference to a POD item like this isn't going to save you anything (`sizeof(double&)` == `sizeof(double)`), so I'd say just hold onto a `const double` here.